test/NuGet.Core.Tests/NuGet.Packaging.Test/SigningTests/ReadOnlyBufferedStreamTests.cs

```diff
@@ -0,0 +1,1076 @@
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(-1)]
+        public void Constructor_ThreeArguments_NonPositiveBufferSize_Throws(int bufferSize)
+        {
+            var exception = Assert.Throws<ArgumentOutOfRangeException>(
+                () => new ReadOnlyBufferedStream(Stream.Null, leaveOpen: false, bufferSize: bufferSize));
+
+            Assert.Equal("bufferSize", exception.ParamName);
+        }
+
+        [Fact]
+        public void Constructor_InitializesSelectProperties()
+        {
+            using (var test = CreateTest())
+            {
+                Assert.Equal(_defaultBuffer.Length, test.BufferedReadStream.Length);
+                Assert.Equal(0, test.BufferedReadStream.Position);
+            }
+        }
+
+        [Fact]
+        public void CanRead_ReturnsValueFromUnderlyingStream()
+        {
+            using (var test = CreateTest())
+            {
+                // The constructor already called CanRead once.
+                Assert.Equal(1, test.UnderlyingStream.CanReadCallCount);
+
+                var canRead = test.BufferedReadStream.CanRead;
+
+                Assert.Equal(2, test.UnderlyingStream.CanReadCallCount);
+                Assert.Equal(test.UnderlyingStream.CanRead, canRead);
+            }
+        }
+
+        [Fact]
+        public void CanSeek_ReturnsValueFromUnderlyingStream()
+        {
+            using (var test = CreateTest())
+            {
+                // The constructor already called CanSeek once.
+                Assert.Equal(1, test.UnderlyingStream.CanSeekCallCount);
+
+                var canSeek = test.BufferedReadStream.CanSeek;
+
+                Assert.Equal(2, test.UnderlyingStream.CanSeekCallCount);
+                Assert.Equal(test.UnderlyingStream.CanRead, canSeek);
+            }
+        }
+
+        [Fact]
+        public void CanTimeout_ReturnsValueFromUnderlyingStream()
+        {
+            using (var test = CreateTest())
+            {
+                Assert.Equal(0, test.UnderlyingStream.CanTimeoutCallCount);
+
+                var canTimeout = test.BufferedReadStream.CanTimeout;
+
+                Assert.Equal(1, test.UnderlyingStream.CanTimeoutCallCount);
+                Assert.Equal(test.UnderlyingStream.CanTimeout, canTimeout);
+            }
+        }
+
+        [Fact]
+        public void CanWrite_ReturnsFalse()
+        {
+            using (var test = CreateTest())
+            {
+                Assert.False(test.BufferedReadStream.CanWrite);
+                Assert.Equal(0, test.UnderlyingStream.CanWriteCallCount);
+            }
+        }
+
+        [Fact]
+        public void Position_WithNegativeValue_Throws()
+        {
+            using (var test = CreateTest())
+            {
+                Assert.Throws<ArgumentOutOfRangeException>(() => test.BufferedReadStream.Position = -1);
+            }
+        }
+
+        [Theory]
+        [InlineData(0)]
+        [InlineData(1)]
+        [InlineData(long.MaxValue)]
```

Comment:
should you be able to set the position of a stream after its length?

Reply:
Yes.